### PR TITLE
fix rxdeep results, add rxjs-hooks

### PR DESCRIPTION
**Pull request by loreanvictor**
follow up to [this PR](https://github.com/dai-shi/will-this-react-global-state-work-in-concurrent-mode/pull/19), fixing RxDeep results (in README) and adding rxjs-hooks for reference.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -626,12 +639,25 @@ in your browser, and click the button very quickly. (check the console log)
...
+
+  <tr>
+    <th><a href="https://github.com/LeetCode-OpenSource/rxjs-hooks">rxjs-hooks</a></th>
+    <td>:white_check_mark:</td>
+    <td>:white_check_mark:</td>
+    <td>:white_check_mark:</td>
+    <td>:white_check_mark:</td>
+    <td>:white_check_mark:</td>
     <td>:x:</td>
+    <td>:white_check_mark:</td>
```
Comment (dai-shi) [inline README.md:658]:
```suggestion
    <td>:grey_question:</td>
```
I think this is false positives. It's the issue in the tool itself.

**Comment by dai-shi** [comment]
Thanks. Merging this. I need to investigate why check 7 doesn't fail with some of rxjs libs.

**Comment by loreanvictor** [comment]
I would love to help, though I would need to read the code more closely to understand what is each check actually checking. The tearing bits are more understandable, but I have no idea what the "EXCEED" error is showing.

**Comment by dai-shi** [comment]
I know the code for check 7 is not easy for reading.

Let's start with some basics.
For state branching, https://reactjs.org/docs/concurrent-mode-intro.html is must read.

Suppose the initial count is `0` and we dispatch the increment action in a transition.
The state is branched and in the first branch it's `0` and in the second branch it's `1`.
While components render with the second branch, we still have the first branch and displayed.
Once the render phase is finished, we show the new tree (rendered with the second branch).

Now, if we dispatch another action outside transition within the pending state, the expected behavior is to apply it to the second branch and do not change the first branch. So, users would see `0` and "Pending..." and then `1`. When they see `2`, no "Pending..."

However, if a global state does not support branching, the second action is applied even if it's in a transition. In this case, users would see `2` with "Pending..."

This is what the current tool does for checking branching. That is `EXCEEDED`. I'm happy to take any other approaches to check branching.

Finally, please note that all of this is what I learned from the doc and the behavior in many experiments. However, there can be some mistakes or React will change the behavior in the future.

**Comment by loreanvictor** [comment]
thanks for the explanation. intuitively makes sense, I'll have to take a deeper look once I got some more free time on my hand.

**Pull request merged**